tests/src/services/KafkaConnectorTests.scala

```diff
@@ -107,4 +112,49 @@ class KafkaConnectorTests extends FlatSpec with Matchers with BeforeAndAfterAll
         }
     }
 
+    it should "catch a failing commit" in {
+        val stream = new ByteArrayOutputStream
+        val printstream = new PrintStream(stream)
+        consumer.outputStream = printstream
+        val messageReceived = "message recieved"
+        try {
+            val message = new Message { override val serialize = Calendar.getInstance().getTime().toString }
+            val sent = Await.result(producer.send(topic, message), 10 seconds)
+            consumer.throwCommitException = true
+            consumer.onMessage((topic, partition, offset, bytes) => {
+                printstream.println(messageReceived)
+            })
+
+            retry(stream.toString should include("exception while pulling new records: commit failed"), 50, Some(100 millisecond))
+            consumer.throwCommitException = false
+
+            // Give the background-thread some time to get the message and commit that it is collected.
+            retry(stream.toString should include(messageReceived), 20, Some(500 millisecond))
+
+            // Wait a few seconds and ensure that the message is not processed twice
+            Thread.sleep(5000)
+            StringUtils.countMatches(stream.toString, messageReceived) should be(1)
+        } finally {
+            printstream.close()
+            stream.close()
+        }
+    }
+
+}
+
+class TestKafkaConsumerConnector(
+    kafkahost: String,
+    groupid: String,
+    topic: String,
+    sessionTimeout: FiniteDuration) extends KafkaConsumerConnector(kafkahost, groupid, topic, sessionTimeout = sessionTimeout) {
+
+    override def commit() = {
+        if (throwCommitException) {
```

Comment:
Take a look at this test in `KafkaConnectorTests: it should "send and receive a kafka message even after session timeout"` as a way of actually getting commit to throw a real exception.